scripts/drs_file_access_examination.ipynb

```diff
@@ -0,0 +1,326 @@
+    "print(str(resolved_DRS_urls_count)+\"/\"+str(len(DRS_urls_dict)) + \" completed... Done\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Create a report\n",
+    "\n",
+    "In this report, we will first print out statistics of the DRS url data. The information record in the statistics is DRS urls that were found in the workspace, that were resolved, that were unresolved. Also, files with and without access. Lastly, errors from DRS urls that are not resolved or files without access.\n",
+    "\n",
+    "The second part of the report is the table of DRS url data that includes the columns: table_name, row_id, drs_url, drs_unresolved_error, file_access_error, file_path, is_resolved, and file_access \n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "\n",
+    "# Checking if there is unresolved_DRS_urls_errors to report\n",
+    "if unresolved_DRS_urls_errors_list:\n",
+    "    unresolved_DRS_urls_errors = \", \".join(set(unresolved_DRS_urls_errors_list))\n",
+    "else:\n",
+    "    unresolved_DRS_urls_errors  = \"N/A\"\n",
+    "    \n",
+    "# Checking if there is files_without_access_errors to report\n",
+    "if files_without_access_errors_list:\n",
+    "    files_without_access_errors = \", \".join(set(files_without_access_errors_list))\n",
+    "else:\n",
+    "    files_without_access_errors  = \"N/A\"\n",
+    "\n",
+    "# Creating the statistics for the reports\n",
+    "print(f'''\n",
+    "_______________________________________________________________________\n",
+    "\n",
+    ":: Data Repository Service (DRS) DRS File Access Examination Report ::\n",
+    "_______________________________________________________________________\n",
+    "\n",
+    "DRS urls found in the workspace: {len(DRS_urls_dict)}\n",
+    "        \n",
+    "DRS urls resolved: {len(DRS_urls_dict)-len(unresolved_DRS_urls_errors_list)}\n",
+    "        \n",
+    "Files with access: {files_with_access}\n",
+    " \n",
+    "DRS urls not resolved: {len(unresolved_DRS_urls_errors_list)}\n",
+    "        \n",
+    "Files without access: {files_without_access}\n",
+    "        \n",
+    "\n",
+    "Errors found from DRS urls that are not resolved: {len(unresolved_DRS_urls_errors_list)}\n",
+    "        \n",
+    "Distinct errors from DRS urls that are not resolved: \n",
+    "        \n",
+    "{unresolved_DRS_urls_errors}\n",
+    "\n",
+    "\n",
+    "Errors found from files without access: {len(files_without_access_errors_list)}\n",
+    "        \n",
+    "Distinct errors from files without access: \n",
+    "        \n",
+    "{files_without_access_errors}\n",
+    "\n",
+    "''')\n",
+    "\n",
+    "# Outputting the table of the master DRS urls dictionary\n",
+    "pd.DataFrame.from_dict(DRS_urls_dict).transpose()\n"
```

Comment:
do you do anything with this dataframe? not sure why this is needed

Reply:
In the notebook, just creating the variable outputs the results